torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3007,10 +3021,164 @@ struct to_ir {
         auto lc = ListComp::create(apply.range(), ident, ident, iter);
         return std::make_shared<SimpleValue>(
             emitListComprehension(lc, type_hint));
+    }
+
+  std::shared_ptr<SugaredValue> emitApplySpecialFormForDict(
+      Symbol& form,
+      Apply& apply,
+      const TypePtr& type_hint = nullptr) {
+    auto treat_as_empty_container = [&]{
+      if (apply.inputs().empty() && !apply.attributes().empty()) {
+        return true;
       }
-      default:
-        TORCH_INTERNAL_ASSERT(false, "unknown special form: ", form);
+      if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_DICT_LITERAL) {
```

Comment:
IMO, we should get rid of the special empty handling - we're not handling dicts/mappings in general, and `dict([])` will work without the special case.